providers/postgres/pyproject.toml

```diff
@@ -61,6 +61,7 @@ dependencies = [
     "apache-airflow-providers-common-sql>=1.23.0",
     "psycopg2-binary>=2.9.9; python_version < '3.13'",
     "psycopg2-binary>=2.9.10; python_version >= '3.13'",
+    "psycopg[binary]>=3.2.9",
```

Comment:
Do we want to have BOTH psycopg2 and pcycopg installed ? I think this shoudl be optional dependency of the provider, especially that we currently don't even use sqla2 and that we have dynamic detection in place.

My proposal would be:

* add psycopg extra and move psycopg there
* add psycopg as dev dependency (for tests)

This will:

* make sure that tests will run on CI with psycopg
* but psycopg will not be installed by default when provider is installed

It's not a huge dependency (200KB) - but it's still considered as experimental/optional

And I would say we can consider switching to it after we run the tests for sqlalchemy 2 for a while and when we finally bump to sqlalechemy 2 as "minimum" version.

Reply:
Thanks for the tip - I've been wondering how to make it optional but still run in tests. Anyway - fixed.